### PR TITLE
[FLINK-10939][doc] Add documents for running Flink cluster natively on Kubernetes

### PR DESCRIPTION
**Pull request by wangyang0918**
## What is the purpose of the change

Add document for running Flink cluster natively on Kubernetes
image

## Brief change log

* Add document

## Verifying this change
```
cd docs
./build_docs.sh
```
View http://localhost:4000/ops/deployment/native_kubernetes.html.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by wangyang0918** [comment]
@aljoscha I have checked all the commands in the doc. Please take a look at your convenience. Thanks a lot.

**Comment by aljoscha** [comment]
Thanks a lot for this addition, @wangyang0918! My (american) colleague @sjwiesman who is a native english speaker looked at the text and suggested some changes. The ordering of the sections was also changed slightly. I pushed a commit with his suggestions. What do you think about them?

**Comment by wangyang0918** [comment]
@aljoscha @sjwiesman Thanks for your review and suggestion. The changes make many sense to me. I just add a fixup and hotfix. Please take a look.

File: docs/ops/deployment/native_kubernetes.zh.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+{% highlight bash %}
+./bin/kubernetes-session.sh \
+  -Dkubernetes.cluster-id=<ClusterId> \
+  -Dtaskmanager.memory.process.size=4096m \
+  -Dkubernetes.taskmanager.cpu=2 \
+  -Dtaskmanager.numberOfTaskSlots=4
+{% endhighlight %}
+
+The system will use the configuration in `conf/flink-conf.yaml`.
+Please follow our [configuration guide]({{ site.baseurl }}/zh/ops/config.html) if you want to change something.
```
Comment (sjwiesman) [inline docs/ops/deployment/native_kubernetes.zh.md]:
```suggestion
Please follow our [configuration guide]({{ site.baseurl }}/ops/config.html) if you want to change something.
```

File: docs/ops/deployment/kubernetes.zh.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ This page describes how to deploy a Flink job and session cluster on [Kubernetes
 * This will be replaced by the TOC
 {:toc}
 
+{% info %} This page describes deploying a [standalone](#cluster_setup.html) Flink session on top of Kubernetes. For information on native Kubernetes deployments read [here]({{ site.baseurl }}/zh/ops/deployment/native_kubernetes.html).
```
Comment (sjwiesman) [inline docs/ops/deployment/kubernetes.zh.md]:
```suggestion
{% info %} This page describes deploying a [standalone](#cluster_setup.html) Flink session on top of Kubernetes. For information on native Kubernetes deployments read [here]({{ site.baseurl }}/ops/deployment/native_kubernetes.html).
```

File: docs/ops/deployment/native_kubernetes.zh.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+Follow these instructions to start a Flink Session within your Kubernetes cluster.
+
+A session will start all required Flink services (JobManager and TaskManagers) so that you can submit programs to the cluster.
+Note that you can run multiple programs per session.
+
+{% highlight bash %}
+$ ./bin/kubernetes-session.sh
+{% endhighlight %}
+
+All the Kubernetes configuration options can be found in our [configuration guide]({{ site.baseurl }}/zh/ops/config.html#kubernetes).
```
Comment (sjwiesman) [inline docs/ops/deployment/native_kubernetes.zh.md]:
```suggestion
All the Kubernetes configuration options can be found in our [configuration guide]({{ site.baseurl }}/ops/config.html#kubernetes).
```

**Comment by wangyang0918** [comment]
@sjwiesman I am not sure whether we could remove `/zh`. Since i find lots of usage `{{ site.baseurl }}/zh` in the doc module and they works well. 

https://ci.apache.org/projects/flink/flink-docs-master/zh/dev/api_concepts.html

**Comment by sjwiesman** [comment]
You are correct, please disregard my comment. +1

**Comment by wangyang0918** [comment]
@sjwiesman Aha, thanks for your quick response. I have checked the zh document in the PR, it works well. 
`./build_docs.sh -i -z`

@aljoscha Please help to squash the first three commits and then merge. Many thanks.

**Comment by aljoscha** [comment]
@sjwiesman Did you try that? When I build the site locally the link to the config section is not prepended with `/zh` if it's not explicitly there in the source file.

**Comment by wangyang0918** [comment]
@aljoscha Yes, we need to add the `/zh` in `{{ site.baseurl }}`. The current behavior is right, i will not update it.

**Comment by sjwiesman** [comment]
@aljoscha I am tired, please ignore.

**Comment by aljoscha** [comment]
😅:coffee: 

I merged this now. Thanks a lot @wangyang0918! And also thanks @sjwiesman for the adjustments.

File: docs/ops/deployment/native_kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+under the License.
+-->
+
+This page describes how to deploy a Flink session cluster natively on [Kubernetes](https://kubernetes.io).
+
+* This will be replaced by the TOC
+{:toc}
+
+<div class="alert alert-warning">
+Flink's native Kubernetes integration is still experimental. There may be changes in the configuration and CLI flags in latter versions. Job clusters are not yet supported.
```
Comment (yanghua) [inline docs/ops/deployment/native_kubernetes.md]: latter -> later?

File: docs/ops/deployment/native_kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+$ ./bin/flink run -d -e kubernetes-session -Dkubernetes.cluster-id=<ClusterId> examples/streaming/WindowJoin.jar
+{% endhighlight %}
+
+### Accessing Job Manager UI
+
+There are several ways to expose a Service onto an external (outside of your cluster) IP address.
+This can be configured using `kubernetes.service.exposed.type`.
+
+- `ClusterIP`: Exposes the service on a cluster-internal IP.
+The Service is only reachable within the cluster. If you want to access the Job Manager ui or submit job to the existing session, you need to start a local proxy.
```
Comment (yanghua) [inline docs/ops/deployment/native_kubernetes.md]: ui -> UI and `submit job` -> `submit a job`?

File: docs/ops/deployment/native_kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+You can then use `localhost:8081` to submit a Flink job to the session or view the dashboard.
+
+{% highlight bash %}
+$ kubectl port-forward service/<ServiceName> 8081
+{% endhighlight %} 
+
+- `NodePort`: Exposes the service on each Node’s IP at a static port (the `NodePort`). `<NodeIP>:<NodePort>` could be used to contact the Job Manager Service. `NodeIP` could be easily replaced with Kubernetes ApiServer address.
+You could find it in your kube config file.
+
+- `LoadBalancer`: Default value, exposes the service externally using a cloud provider’s load balancer.
```
Comment (yanghua) [inline docs/ops/deployment/native_kubernetes.md]: `Default value, exposes` -> `Default value exposes`?

File: docs/ops/deployment/native_kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+Since the cloud provider and Kubernetes needs some time to prepare the load balancer, you may get a `NodePort` JobManager Web Interface in the client log.
+You can use `kubectl get services/<ClusterId>` to get EXTERNAL-IP and then construct the load balancer JobManager Web Interface manually `http://<EXTERNAL-IP>:8081`. 
+
+- `ExternalName`: Map a service to a DNS name, not supported in current version.
```
Comment (yanghua) [inline docs/ops/deployment/native_kubernetes.md]: `in current version` -> `in the current version`?

File: docs/ops/deployment/native_kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+{% highlight bash %}
+$ echo 'stop' | ./bin/kubernetes-session.sh -Dkubernetes.cluster-id=<ClusterId> -Dexecution.attached=true
+{% endhighlight %}
+
+#### Manual Resource Cleanup
+
+Flink uses [Kubernetes ownerReference's](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/) to cleanup all cluster components.
+All the Flink created resources, including `ConfigMap`, `Service`, `Deployment`, `Pod`, have been set the ownerReference to `service/<ClusterId>`. 
+When the service is deleted, all other resource will be deleted automatically.
```
Comment (yanghua) [inline docs/ops/deployment/native_kubernetes.md]: resource -> resources

**Pull request closed without merging**